### ISSUE TITLE
Axe is reporting 'Elements must have sufficient color contrast' for input

### ISSUE THREAD

**Issue by Kretajak** [JedWatson/react-select#4814]
Axe is reporting `Elements must have sufficient color contrast` with comment: `Element's background color could not be determined because it is overlapped by another element`. 

This is because `<input>` element is covered by `placeholder` element thus not letting axe to calculate contrast. This can block axe from catching violations like a too low contrast.

**Steps to reproduce:**

* Enter page: https://798fh.csb.app/
* Run axe check

**Workaround**:

Use custom CSS that will render input in front of absolutely positioned placeholder: 

```scss
div.placeholder {
  & + div {
    position: relative;

    input {
        cursor: pointer;
    }
  }
}
```

**Comment by ebonow** [comment]
@Kretajak this error seems very much like a red herring. 

The intention of the placeholder is to be exactly that. It is some text that represents what the input placeholder should be. It is rendered because the Input component's opacity is set to 0 when the control loses focus. 

When there is any input, the Placeholder component is removed from the DOM so there would not be any conflict in contrast.

My desire is to deprecate this component entirely so that the input's placeholder attribute can be used instead. This would additionally resolve some other css complications, but that's another discussion entirely. That all said, I think it's important when discussing these things if it is about a11y UX or checking off some validator checkbox. In this case, my argument is that the context of the situation accounts for the "hypothetical" conflict that AXE is checking for.

In any case, here is an implementation I use that replaces the Placeholder component with the actual input placeholder attribute. For context, I pass in a floating label which displays over where the "Placeholder" component normally would but is positioned absolutely. This allows me to have a label and an actual placeholder both displayed when the input is focused.

``` jsx
const Input = (props) => {
  const { placeholder, label, maxLength } = props.selectProps;
  let inputProps = { ...props, maxLength };

  if (!!placeholder && !!label) {
    inputProps.onFocus = e => {
      const target = e.target;
      setTimeout(() => { target.placeholder = placeholder }, 200);
      //https://bugs.chromium.org/p/chromium/issues/detail?id=468153#c164
      target.autocomplete = 'chrome-off';
      props.onFocus(e);
    };

    inputProps.onBlur = e => {
      const target = e.target;
      
      e.target.placeholder = '';
      // In case blur does not happen before onFocus timeout
      setTimeout(() => { target.placeholder = '' }, 200); 
      
      props.onBlur(e);
    };
  }
  
  return (
    <components.Input {...inputProps} />
  );
};

return <Select components={{ Input, Placeholder: () => {} }} {...otherProps} />
```
